File: app/src/main/java/com/vkpapps/soundbooster/utils/IPManager.java
package com.vkpapps.soundbooster.utils;

import android.content.Context;
import android.net.DhcpInfo;
import android.net.wifi.WifiManager;

import static android.content.Context.WIFI_SERVICE;

public class IPManager {
    private WifiManager manager;

    public IPManager(Context context) {
        this.manager = (WifiManager) context.getApplicationContext().getSystemService(WIFI_SERVICE);
    }

    private String intToIp(int i) {
        return (i & 0xFF) + "." +
                ((i >> 8) & 0xFF) + "." +
                ((i >> 16) & 0xFF) + "." +
                ((i >> 24) & 0xFF);
    }

    public String hostIp() {
        DhcpInfo dhcpInfo = manager.getDhcpInfo();
        return intToIp(dhcpInfo.gateway);
    }


}

File: app/src/main/java/com/vkpapps/soundbooster/utils/IPManager.kt
package com.vkpapps.soundbooster.utils

import android.content.Context
import android.net.wifi.WifiManager

/**
 * @author VIJAY PATIDAR
 */
class IPManager(context: Context) {
    private val manager: WifiManager = context.applicationContext.getSystemService(Context.WIFI_SERVICE) as WifiManager

    /**
     * @param i take int ip address
     * @return string form of ip
     */
    private fun intToIp(i: Int): String {
        return (i and 0xFF).toString() + "." + (i shr 8 and 0xFF) + "." +
                (i shr 16 and 0xFF) + "." + (i shr 24 and 0xFF)
    }

    fun hostIp(): String {
        return intToIp(manager.dhcpInfo.gateway)
    }
}